### PR TITLE
[quant] Serial-correlation-robust (HAC) Deflated Sharpe SE on the rigor gate (#621) !minor

### PR DESCRIPTION
**Pull request by onder-akkaya**
## Summary

The Tier-1 rigor gate computes a Deflated Sharpe Ratio (Bailey & López de Prado 2014) whose
z-statistic denominator is the **IID** asymptotic variance of the Sharpe estimator. That
standard error is only valid for serially independent returns. Under autocorrelation it is
biased — positive serial dependence *understates* it and *inflates* significance (Lo 2002).

We already detect this: the IID diagnostic suite (Ljung–Box, Lo–MacKinlay variance-ratio,
runs test) was wired in under #621 — but **as an advisory only**. So the gate was in the
self-contradictory position of computing a verdict that says *"this Sharpe SE may be
wrong"* and then gating on exactly that SE. This PR closes that loop: the gate now uses a
**serial-correlation-robust (HAC) standard error**, so a detected autocorrelation is
*corrected in the inference*, not merely flagged.

This is the statistically-correct resolution of #621 — neither a discarded advisory, nor a
hard fifth gate (which would wrongly reject the autocorrelation that *is* the edge for
trend/momentum strategies).

## The correction (derivation)

The Sharpe-ratio estimator's influence function (delta method on the first two sample
moments — Casella & Berger 2002, §10.1.2; Lo 2002; Mertens 2002) is, with standardized
excess return `z_t = (x_t − μ)/σ`:

```
IF_t = z_t − (ŜR/2)(z_t² − 1)
```

Its IID variance is **exactly** the existing Bailey-LdP denominator:

```
Var(IF_t) = 1 − γ₃·ŜR + ((γ₄−1)/4)·ŜR²        (γ₄ = raw/Pearson kurtosis)
```

The heteroskedasticity- and autocorrelation-consistent (HAC) variance replaces this with the
**Newey–West (1987) long-run variance** of `IF_t`, using a Bartlett kernel:

```
LRV = γ₀ + 2·Σ_{k=1}^{L} (1 − k/(L+1))·γ_k ,   γ_k = Cov(IF_t, IF_{t−k})
```

Four properties make this a safe, principled drop-in:

1. **It nests IID.** Under no serial dependence `γ_k → 0 (k ≥ 1)`, so `LRV → γ₀ ≈ Var(IF_t)`.
   `hac_lags=None` reproduces the prior behaviour byte-for-byte; every existing caller is
   unchanged.
2. **Correct direction.** Positive autocorrelation ⇒ fewer independent observations than `T`
   ⇒ `LRV > γ₀` ⇒ wider SE ⇒ smaller `z` ⇒ **stricter** gate. Mean-reversion does the
   reverse. (For an AR(1), `LRV/γ₀ → (1+ρ)/(1−ρ)`.)
3. **Non-negativity for free.** The Bartlett kernel is the Fejér kernel up to scale, so the
   estimator is guaranteed PSD (Newey & West 1987) — no ad-hoc variance flooring.
4. **No pre-test bias.** The gate applies the HAC SE **unconditionally** rather than swapping
   SEs only when the IID flag fires; a conditional (pre-test) estimator has a distorted
   sampling distribution (Leeb & Pötscher 2005). Bandwidth is the Newey–West (1994) plug-in
   `L = ⌊4·(T/100)^(2/9)⌋` (= 4 at T = 252).

## Scope

- **`backend/archimedes/services/_rigor_helpers.py`**
  - `_sharpe_influence_lrv()` — NW-HAC long-run variance of the Sharpe influence function.
  - `_nw_auto_bandwidth()` — Newey–West (1994) automatic lag.
  - `compute_dsr(..., hac_lags=None)` and `_dsr_from_stats(..., variance_override=None)` —
    both default to the exact prior IID closed form.
- **`backend/archimedes/services/rigor_evaluator.py`**
  - `run_rigor_gate` computes the gating DSR with `hac_lags="auto"`, and records the IID-SE
    p-value + the SE method on `RigorGateResult`.
  - `gate_details` gains a `"dsr_se"` key disclosing the SE model and the **IID → HAC delta**
    (e.g. `IID-SE p=0.987 → HAC p=0.961 (Δ=−0.026)`), so the passport is honest about how
    much the correction moved the verdict; the IID advisory string now states the SE is
    HAC-corrected.

## Acceptance criteria

- [x] `pytest backend/tests/test_dsr_hac.py -q` → **13 passed** (nesting, variance-inflation
      direction at the LRV level, NW bandwidth, Bartlett PSD, edge cases, gate surface).
- [x] `pytest backend/tests/test_rigor_evaluator.py backend/tests/test_dsr_parity.py backend/tests/test_rigor_regime.py backend/tests/test_selection_bias_routes.py backend/tests/test_sharpe_statistics.py backend/tests/test_rigor_deploy_gate_contract.py backend/tests/test_return_diagnostics.py -q` → **289 passed**.
- [x] Downstream `compute_dsr` consumers (backtester / stockbench / pipeline) → **59 passed**, default path unaffected.
- [x] `ruff format --check` + `ruff check --select E9,F63,F7,F40` clean on all changed files.

## Verify

```bash
pytest backend/tests/test_dsr_hac.py -q
pytest backend/tests/test_rigor_evaluator.py -q
ruff format --check backend/archimedes/services/_rigor_helpers.py backend/archimedes/services/rigor_evaluator.py
```

## Anti-goals

- The default `compute_dsr(...)` path (`hac_lags=None`) is **unchanged** — no existing caller
  shifts; only `run_rigor_gate` opts into HAC.
- IID is **not** promoted to a hard pass/fail criterion — it remains advisory; the *inference*
  is corrected instead.
- No change to the multiple-testing deflation (`num_trials`, effective-N) — orthogonal to this.

## Note for reviewers

This builds directly on the IID diagnostic wired in **#621** and sits alongside the DSR
multiple-testing deflation in **#811 / #770**. It is a behaviour change on the live gate
(the gating p-value now rests on the HAC SE), so I've surfaced the IID-vs-HAC delta on the
passport to keep the change auditable rather than silent. @dbrowneup @mnemonik-dev — flagging
for your eyes on the rigor-gate path, since #621 and #811 are adjacent; happy to walk through
the derivation. The full influence-function derivation and the Newey–West nesting were checked
independently before commit.

## References

- Bailey, D. & López de Prado, M. (2014). *The Deflated Sharpe Ratio.* J. Portfolio Mgmt 40(5).
- Lo, A. (2002). *The Statistics of Sharpe Ratios.* Financial Analysts Journal 58(4).
- Mertens, E. (2002). *Comments on the correct variance of estimated Sharpe ratios.*
- Newey, W. & West, K. (1987). *A Simple, Positive Semi-Definite, Heteroskedasticity and
  Autocorrelation Consistent Covariance Matrix.* Econometrica 55(3).
- Newey, W. & West, K. (1994). *Automatic Lag Selection in Covariance Matrix Estimation.*
  Review of Economic Studies 61(4).
- Leeb, H. & Pötscher, B. (2005). *Model Selection and Inference: Facts and Fiction.*
  Econometric Theory 21(1).
- Casella, G. & Berger, R. (2002). *Statistical Inference* (2nd ed.), §10.1 (asymptotic
  evaluations, the delta method).

### CODE REVIEW

File: backend/archimedes/services/rigor_evaluator.py [hunk 1]
```diff
@@ -564,9 +590,21 @@ def run_rigor_gate(
...
+    #    influence function, which NESTS the IID form (a no-op on serially
+    #    independent returns) and avoids the pre-test distortion of conditionally
+    #    swapping SEs only when the IID flag fires (Leeb & Pötscher 2005). The
+    #    effective-N correction (average_correlation) still relaxes the
+    #    multiple-testing penalty when the trials themselves are correlated.
+    deflated_sharpe, dsr_p_value = compute_dsr(daily_returns, num_trials, average_correlation, hac_lags="auto")
+    # Advisory companion: the IID-SE p-value, surfaced as a delta (never gates)
+    # so the passport shows how much the serial-dependence correction moved the
+    # verdict relative to the classical Bailey-LdP IID standard error.
+    _, dsr_p_value_iid = compute_dsr(daily_returns, num_trials, average_correlation)
```
[the remark this answers is not in the document]
Comment (onder-akkaya) [reply backend/archimedes/services/rigor_evaluator.py]: Good catch — fixed in 48f4997. Factored the shared series-validation + moment extraction (numpy coercion, SciPy skew/kurtosis) into `_sharpe_dsr_inputs()`, and added `compute_dsr_hac_and_iid()` which returns `(dsr_hac, p_hac, dsr_iid, p_iid)` from a single moment computation + single influence-function LRV. `run_rigor_gate` now calls it once instead of `compute_dsr` twice. A parity test pins that the single-pass result is bit-for-bit equal to the two separate calls, so it's a pure hot-path saving with no numerical change.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
